src/test/ui/inference/need_type_info/expr-struct-type-relative-gat.stderr

```diff
@@ -0,0 +1,9 @@
+error[E0282]: type annotations needed
+  --> $DIR/expr-struct-type-relative-gat.rs:17:9
+   |
+LL |         Self::Output::Simple {};
+   |         ^^^^^^^^^^^^ cannot infer type for type parameter `T` declared on the associated type `Output`
```

Comment:
We should point at `Output` as well.

Reply:
I meant something like

```
error[E0282]: type annotations needed
  --> $DIR/expr-struct-type-relative-gat.rs:17:9
   |
LL |     type Output<T> = Bar<T>;
   |                 - type parameter `T` that couldn't be infered on `Output`
LL |     fn baz() {
LL |         Self::Output::Simple {};
   |         ^^^^^^^^^^^^ cannot infer type for type parameter `T` declared on the associated type `Output`
```

as a stop gap solution, but ideally all of this would look like this

```
error[E0282]: type annotations needed
  --> $DIR/expr-struct-type-relative-gat.rs:17:9
   |
LL |     type Output<T> = Bar<T>;
   |                 - type parameter `T` that couldn't be infered on `Output`
LL |     fn baz() {
LL |         Self::Output::Simple {};
   |         ^^^^^^^^^^^^ cannot infer type for type parameter `T` declared on the associated type `Output`
help: specify the type parameter on associated type `Output` 
   |
LL |         Self::Output::<T>::Simple {};
   |                     +++++
```